worker/draft.go

```diff
@@ -2038,6 +2038,15 @@ func (n *node) calculateSnapshot(startIdx, lastIdx, minPendingStart uint64) (*pb
 					maxCommitTs = x.Max(maxCommitTs, txn.CommitTs)
 				}
 			}
+
+			// If there is a restore proposal then consider the restoreTs as commitTs for the
```

Comment:
This looks good to me. Also, it would be better to add a comment why the `restoreTs` can be taken as `commitTs` in this case. Because this is against the normal case scenario where commits come via delta proposals.

As per my understanding, this is because restoreTs provides the guarantee that all proposals before the restoreTs would be committed already.